File: HexAssetAPI/Models/AssetAllocation.cs
using System.ComponentModel.DataAnnotations;

namespace HexAsset.Models
{
	public class AssetAllocation
	{
		[Key]
		public int AllocationId { get; set; }

		public int AssetId { get; set; }
		public int UserId { get; set; }
		public DateTime AllocationDate { get; set; } = DateTime.UtcNow;
		public DateTime? ReturnDate { get; set; }
		public required string AllocationStatus { get; set; }


		public Asset? Asset { get; set; }
		public User? User { get; set; }
	}
}

File: HexAssetUI/src/app/models/AssetAllocation.ts
export class AssetAllocationModel {
    allocationId: number;
    assetId: number;
    userId: number;
    allocationDate: Date;
    returnDate: Date;
    allocationStatus: string;


    constructor() {
        this.allocationId = 0;
        this.assetId = 0;
        this.userId = 0;
        this.allocationDate = new Date();
        this.returnDate = new Date();
        this.allocationStatus = "";
    }
}